lib/markdown2.py

```diff
@@ -390,8 +390,16 @@ def convert(self, text):
 
         text += "\n"
 
+        # Insert TOC HTML into the MD HTML after the first heading element if any headings are present.
+        if ("inline-toc" in self.extras and self._toc[0] is not None):
+            (level, id, name) = self._toc[0]
+            # Use a regex and rely on the HTML structure as opposed to tracking the heading regex's `end()` across all the HTML transformations (unreliable)
+            # TODO (Tomas Hubelbauer): Consider looser regex which allows for more attributes in order to to find heading even when more extras add attributes to it (future-proof)
```

Comment:
I am considering not doing that now because it feels premature. @nicholasserra thoughts?